### DISCUSSION TITLE
Obtaining Serialized XML in Authentication Hook

### DISCUSSION THREAD

**Discussion by ahmayun** [schemathesis/schemathesis#2339] [Q&A] [answered]
I'm trying to run schemathesis on a service that requires requests to be signed before they are sent. 
I have implemented most of it, however, the signing algorithm requires me to compute the sha256 hash of the request payload. The problem is that if the request payload is xml, I don't seem to have access to the final (serialized) XML that will be sent to the server. Am I missing something. Is there a way to obtain this?

**Comment by Stranger6667** [comment, marked as answer]
Hi!

I think you can use `case.as_transport_kwargs` and then check the `data` field in the returned dictionary. Let me know if it is sufficient

**Reply by ahmayun** [reply]
Worked like a charm! Thank you.

**Reply by Stranger6667** [reply]
Awesome!
